### ISSUE TITLE
Chained type alias assignment for Matrix breaks mypy in SymPy 1.10.1

### ISSUE THREAD

**Issue by agryman** [sympy/sympy#23647]
I was using type hints like:

M: Matrix = Matrix(2)

Using SymPy 1.9, mypy did not report any errors. I then upgraded to SymPy 1.10.1 (which enabled mypy type checking) and mypy started reporting "not valid as a type" errors. 

The root cause was identified by @oscarbenjamin as a mypy limitation in handling chained type alias assignements. 

Matrix = MutableMatrix = MutableDenseMatrix

I reported this problem to the mypy project. Refer to https://github.com/python/mypy/issues/12995

Here is a simple test case to reproduce the error:

----- begin -------
from sympy import Matrix

M: Matrix = Matrix(2)
------- end --------

mypy reports the following:

$ mypy bug_sympy_matrix.py 
bug_sympy_matrix.py:30: error: Variable "sympy.matrices.Matrix" is not valid as a type
bug_sympy_matrix.py:30: note: See https://mypy.readthedocs.io/en/stable/common_issues.html#variables-vs-type-aliases
Found 1 error in 1 file (checked 1 source file)

See attached
[bug_sympy_matrix.py.txt](https://github.com/sympy/sympy/files/8934269/bug_sympy_matrix.py.txt)
.

**Comment by asmeurer** [comment]
Somewhat related, we may at some point want to change `Matrix` from being simply an alias for `MutableDenseMatrix` to a separate class so that we can deprecate default mutability (https://github.com/sympy/sympy/issues/6509). I suspect that such a class would not work with mypy unless we explicitly made it do so, at least for the deprecation period.

**Comment by agryman** [comment]
I opened issue https://github.com/python/mypy/issues/12995 to report this problem to the mypy. They closed it as "won't fix" and as a duplicate of a similar issue that occurs with parallel assignment of type aliases. They also cited PEP 613.

After rereading PEP 484, I think the mypy folks have a legitimate point.

See https://peps.python.org/pep-0484/#type-aliases

"Type aliases are defined by **simple** variable assignments:"

See also https://peps.python.org/pep-0613/#abstract

"Type aliases are user-specified types which may be as complex as any type hint, and are specified with a **simple** variable assignment on a module top level."

In both PEPs I've highlighted the term **simple**.

In summary, Python has several flavours of assignment: simple, chained, parallel
The PEPs state that type aliases are defined by **simple** assignments.

I will open a PR to split the chained assignment for Matrix. I'll also check the code for other instances of non-simple type alias assignments.

**Comment by agryman** [comment]
I am going to fix this.
